core/src/main/java/org/neo4j/graphalgo/api/CypherGraphSetup.java

```diff
@@ -0,0 +1,131 @@
+package org.neo4j.graphalgo.api;
+
+import org.neo4j.kernel.internal.Version;
+
+import java.util.concurrent.ExecutorService;
+
+/**
+ * @author mh
+ * @since 27.04.17
+ */
+public class CypherGraphSetup implements GraphDefinition {
```

Comment:
I'm not sure if we really need the two implementations.
If we add the two statement properties to the GraphLoader API, we can keep the one impl GraphSetup/Definition situation, have the CyperFactory extends GraphFactory as well and use the same API for all loaders, which would allow us to include it easily in tests like GraphNegativeTest.
Since loading only happens once, we can do some validation, maybe throw an IllegalState|ArgumentException, when defining both, Label/RelTypes and Cypher Statements.

WDYT?